### PR TITLE
[public-api] Implement UpdateTeamMember and DeleteTeamMember

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Implements updating team member (setting role) and deleting team members.

## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/14311
* Part of https://github.com/gitpod-io/gitpod/issues/14312

## How to test

Unit tests
You can see this implementation in https://github.com/gitpod-io/gitpod/pull/14412, follow test instructions

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/gitpod-protocol/go/gitpod-service.go [hunk 1]
```diff
@@ -1481,6 +1487,26 @@ func (gp *APIoverJSONRPC) ResetGenericInvite(ctx context.Context, teamID string)
...
+func (gp *APIoverJSONRPC) RemoveTeamMember(ctx context.Context, teamID, userID string) (err error) {
+	if gp == nil {
+		err = errNotConnected
+		return
+	}
+	_params := []interface{}{teamID, userID}
+	err = gp.C.Call(ctx, string(FunctionRemoveTeamMember), _params, nil)
+	return
+}
+
```
Comment (andrew-farries) [inline components/gitpod-protocol/go/gitpod-service.go:1509]: How come this is necessary? `setTeamMemberRole` and `RemoveTeamMember` are already on `server` and presumably were being used, so why do they need to be added here?

**Review by andrew-farries — APPROVED** [review]
One question

File: components/gitpod-protocol/go/gitpod-service.go (hunk 1 above)
Comment (easyCZ) [reply components/gitpod-protocol/go/gitpod-service.go:1509]: They were on server, but the Go interface didn't have them. It's not auto-generated, it's hand-coded and there's nothing to ensure the interfaces match :/

**Pull request merged**